File: pyg_parent/pyg_manager/pyg_manager_interface/src/main/java/com/pyg/service/BrandService.java
package com.pyg.service;



import com.pyg.pojo.PageResult;
import com.pyg.pojo.PygResult;
import com.pyg.pojo.TbBrand;

import java.util.List;

/**
 * @author mayday
 */
public interface BrandService {

    /**
     * 分页查询查询全部产品
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult<TbBrand> listAllBrand(int pageNum, int pageSize);

    /**
     * 保存品牌的方法
     * @param brand
     * @return
     */
    void saveBrand(TbBrand brand);

    /**
     * 根据 id 删除品牌
     * @param ids
     */
    void deleteBrands(long[] ids);

    /**
     * 根据 id 查询单个品牌
     * @param id
     * @return
     */
    TbBrand findBrandById(long id);

    /**
     * 修改品牌
     * @param brand
     */
    void updateBrand(TbBrand brand);

    /**
     * 根据条件分页查询品牌
     * @param pageNum
     * @param pageSize
     * @param searchBrand
     * @return
     */
    PageResult<TbBrand> searchBrandList(int pageNum, int pageSize, TbBrand searchBrand);
}

File: pyg_parent/pyg_manager_web/src/main/webapp/admin/js/service/brandService.js
app.service("brandService", function ($http) {

    this.save = function ($scope) {
        $scope.method = "saveBrand";
        if ($scope.brand.id != null) {
            $scope.method = "updateBrand";
        }

        return $http.post('../brand/' + $scope.method, $scope.brand);
    };

    this.delete = function ($scope) {
        return $http.get('../brand/deleteBrands?ids=' + $scope.selectIds);
    };

    this.findBrandById = function (id) {
        return $http.get('../brand/findBrandById/' + id);
    };

    this.search = function (pageNum, pageSize, searchBrand) {
        if (pageNum == null){
            pageNum = 1
        }
        if (pageSize == null) {
            pageSize = 10
        }

        return $http.post('../brand/searchBrandList?pageNum=' + pageNum + '&pageSize=' + pageSize, searchBrand)
    };

})